include/network_interface/network_utils.h

```diff
@@ -56,24 +56,48 @@ T read_le(const std::vector<uint8_t>& bufArray,
 };
 
 template<typename T>
-std::vector<uint8_t> write_le(T *source)
+std::vector<uint8_t> write_le(T *source,
+    typename std::enable_if<std::is_integral<T>::value>::type* = 0)
 {
   std::vector<uint8_t> ret_val;
 
-  if (std::is_floating_point<T>::value)
-    return ret_val;
-
   T mask = 0xFF;
+  uint32_t shift = 0;
 
-  while ((*source & mask) > 0)
+  while (shift < (8 * sizeof(T)))
```

Comment:
8 meaning 8 bits or bytes.  It seems trivial, but using #defines for these numbers is helpful for quick comprehension.

Reply:
@srustanAStuff In this context, it can only mean bits. `sizeof()` returns a number of bytes and `shift` represents a number of bits by which to shift a value. To convert from the return value of `sizeof()` to a number of bits to shift, you have to multiply by 8. Replacing the number 8 with any `#define` would be more verbose than necessary here (e.g. `BITS_IN_BYTE`). Readability is important but a certain level of code comprehension is assumed.